src/main/java/io/jenkins/plugins/pipelinegraphview/utils/AbstractPipelineViewAction.java

```diff
@@ -56,19 +61,37 @@ public String getBuildDisplayName() {
      */
     @RequirePOST
     @JavaScriptMethod
-    public boolean doRebuild() throws IOException, ExecutionException {
+    public JSONObject doRebuild() throws IOException, ExecutionException, Exception {
+        JSONObject result = new JSONObject();
+        result.put("success",false);
+        result.put("url","");
+
         if (run != null) {
             run.checkAnyPermission(Item.BUILD);
             ReplayAction replayAction = run.getAction(ReplayAction.class);
             Queue.Item item =
                     replayAction.run2(replayAction.getOriginalScript(), replayAction.getOriginalLoadedScripts());
 
             if (item == null) {
-                return false;
+                return result;
             }
-            return true;
+
+            QueueTaskFuture<? extends Run<?, ?>> f = (QueueTaskFuture) item.getFuture();
```

Comment:
can you have a look at how blue ocean does it please?

I don't think we want a thread waiting for start.

A better way is probably to poll in the frontend for a bit?

Reply:
@timja Yes I will do it